client/logmon/logging/rotator.go

```diff
@@ -54,7 +54,7 @@ type FileRotator struct {
 	doneCh      chan struct{}
 
 	closed     bool
-	closedLock sync.Mutex
+	closedLock sync.Mutex // also used to guard oldestLogFileIdx
```

Comment:
Might be worth giving the lock a general name and moving the locked values under it